lfsapi/ssh.go

```diff
@@ -124,34 +124,44 @@ func sshFormatArgs(cmd string, args []string, needShell bool) (string, []string)
 	return "sh", []string{"-c", joined}
 }
 
-func sshGetLFSExeAndArgs(osEnv config.Environment, e Endpoint, method string) (string, []string) {
-	exe, args, needShell := sshGetExeAndArgs(osEnv, e)
+func sshGetLFSExeAndArgs(osEnv config.Environment, gitEnv config.Environment, e Endpoint, method string) (string, []string) {
+	exe, args, needShell := sshGetExeAndArgs(osEnv, gitEnv, e)
 	operation := endpointOperation(e, method)
 	args = append(args, fmt.Sprintf("git-lfs-authenticate %s %s", e.SshPath, operation))
 	exe, args = sshFormatArgs(exe, args, needShell)
 	tracerx.Printf("run_command: %s %s", exe, strings.Join(args, " "))
 	return exe, args
 }
 
+// Parse command, and if it looks like a valid command, return the ssh binary
+// name, the command to run, and whether we need a shell.  If not, return
+// existing as the ssh binary name.
+func sshParseShellCommand(command string, existing string) (ssh string, cmd string, needShell bool) {
+	ssh = existing
+	cmdArgs := tools.QuotedFields(command)
```

Comment:
Nit: to limit the scope of `cmdArgs`, you could also write:

```go
if cmdArgs := tools.QuotedFields(command); len(cmdArgs) > 0 {
        // ...
}
```

But I don't suspect that this function will change much in the future, so I think that you'd be equally fine to ignore this suggestion.

Reply:
Yeah, that would be a nice improvement.  I’ll fix and resubmit.